src/transformers/models/whisper/feature_extraction_whisper.py

```diff
@@ -301,23 +474,37 @@ def __call__(
 
         # convert into correct format for padding
 
+        # todo: auto return_attention_mask
         padded_inputs = self.pad(
             batched_speech,
             padding=padding,
             max_length=max_length if max_length else self.n_samples,
             truncation=truncation,
             pad_to_multiple_of=pad_to_multiple_of,
+            return_attention_mask=True,
         )
         # make sure list is in array format
         input_features = padded_inputs.get("input_features").transpose(2, 0, 1)
 
+        # mono
         input_features = [self._np_extract_fbank_features(waveform) for waveform in input_features[0]]
 
         if isinstance(input_features[0], List):
             padded_inputs["input_features"] = [np.asarray(feature, dtype=np.float32) for feature in input_features]
         else:
             padded_inputs["input_features"] = input_features
 
+        # todo: move to config
+        apply_spec_augment = True
+        if apply_spec_augment:
+            # todo: input_features to np array
+            padded_inputs["input_features"] = np.stack(padded_inputs["input_features"], 0)
+
+            padded_inputs["input_features"] = self._mask_input_features(
+                padded_inputs["input_features"],
+                attention_mask=padded_inputs.attention_mask[:, ::self.hop_length],
+            )
```

Comment:
Let's make sure that we don't change the return type here! (previously was either a List or a `np,array`

Reply:
`padded_inputs["input_features"]` was previously a `List[np.array]`, and will be casted into tensor in the line 509.

Could we make `padded_inputs["input_features"]` always be `np.array` using or not using spec_augment? Since `self._mask_input_features` always accepts and returns `np.array`

Reply:
Hi @ArthurZucker , I think this PR is about to finish except this. What's your opinion here :)